web/pandas/pdeps/0007-copy-on-write.md

```diff
@@ -0,0 +1,575 @@
+
+On the other hand, the proposal in this document does not give the user control over
+whether a subset should be a view (when possible) that mutates the parent when being
+mutated. The only way to modify the parent dataframe is with a direct indexing operation
+on this dataframe itself.
+
+See the GitHub comment with some more detailed argumentation:
+[https://github.com/pandas-dev/pandas/issues/36195#issuecomment-786654449](https://github.com/pandas-dev/pandas/issues/36195#issuecomment-786654449)
+
+## Disadvantages
+
+Other than the fact that this proposal would result in a backwards incompatible,
+breaking change in behaviour (see next section), there some other potential
+disadvantages:
+
+* Deviation from numpy: numpy uses the copy and view concepts, while in this proposal
+  views would basically not exist anymore in pandas (for the user, at least; we would
+  still use it internally as an implementation detail)
+  * But as a counter argument: many pandas users are probably not familiar with those
+    concepts, and pandas already deviates from the exact rules in numpy.
+* Performance cost of indexing and methods becomes harder to predict: because the copy
+  of the data doesn't happen at the moment when actually creating the new object, but
+  can happen at a later stage when modifying either the parent or child object, it
+  becomes less transparent about when pandas copies data (but in general we should copy
+  less often).
+* Increased memory usage for some use cases: while the majority of use cases will
+  probably see an improvement in memory usage with this proposal, there are a few use
+  cases where this might not be the case. Specifically in cases where pandas currently
+  does return a view (eg slicing rows) and in the case you are fine with (or don't care
+  about) the current behaviour of it being a view when mutating that subset (i.e.
+  mutating the sliced subset also mutates the parent dataframe), in such a case the
+  proposal would introduce a new copy compared to the current behaviour.
+
+## Backward compatibility
+
+The proposal in this document is clearly a backwards incompatible change that breaks
+existing behaviour. Because of the current inconsistencies and subtleties around views
+vs. copies and mutation, it would be difficult to change anything without breaking
+changes. The current proposal is not the proposal with the minimal changes, though. A
+change like this will in any case need to be accompanied with a major version bump (for
+example pandas 3.0).
+
+Doing a traditional deprecation cycle that lives in several minor feature releases will
+probably be too noisy. Indexing is too common an operation to include a warning (even if
+we limit it to just those operations that previously returned views). However, we can
+already implement this proposal and make it available before becoming the default
+behaviour, such that users can opt-in and test their code (this is currently possible
```

Comment:
might want to say "this is possible as of version 1.?" instead of "currently"

Reply:
Yep